android/src/main/kotlin/in/juspay/hyper_webview_flutter/HyperWebviewFlutterPlugin.kt

```diff
@@ -4,39 +4,155 @@
-  /// when the Flutter Engine is detached from the Activity
-  private lateinit var channel : MethodChannel
-
-  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
-    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "hyper_webview_flutter")
+class HyperWebviewFlutterPlugin: FlutterPlugin, MethodChannel.MethodCallHandler, ActivityAware, PluginRegistry.ActivityResultListener{
+  private var binding: ActivityPluginBinding? = null
+
+  private lateinit var channel: MethodChannel
+  private lateinit var dartChannel : MethodChannel;
+
+  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "NativeChannel")
     channel.setMethodCallHandler(this)
+    dartChannel =  MethodChannel(flutterPluginBinding.binaryMessenger, "DartChannel");
+  }
+
+  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+    channel.setMethodCallHandler(null)
   }
 
-  override fun onMethodCall(call: MethodCall, result: Result) {
-    if (call.method == "getPlatformVersion") {
-      result.success("Android ${android.os.Build.VERSION.RELEASE}")
-    } else {
-      result.notImplemented()
+  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: MethodChannel.Result) {
+    var args = call.arguments<List<String>>();
+    try {
+      when (call.method) {
+        "findApps" -> {
+          var payload = args?.first();
+          this.binding?.activity?.let {
+            var allApps = UPIInterface.findApps(it, payload);
+            var output = hashMapOf<String, Any>();
+            output["requestCode"] = Constants.FINDAPPS_REQUEST_CODE;
+            output["payload"] = allApps;
+            dartChannel.invokeMethod("onActivityResult", output);
+          }
+        }
+
+        "openApp" -> {
+          var args = call.arguments<List<String>>();
+          var packageName = args!![0];
+          var payload = args!![1];
+          var action = args!![2];
+          var flag = args!![1].toIntOrNull() ?: 0 // whats the correct default?
+          this.binding?.activity?.let { UPIInterface.openApp(it, packageName, payload, action, flag) };
+        }
+        "getResourceByName" -> {
+          var args = call.arguments<List<String>>();
+          var resName = args!![0];
+          this.binding?.activity?.let {
+             var resourceVal = UPIInterface.getResourceByName(it, resName);
+             var output = hashMapOf<String, Any>();
+             output["requestCode"] = Constants.GET_RESOURCE_NAME;
+             output["payload"] = resourceVal;
+             dartChannel.invokeMethod("onActivityResult", output);
+           };
+        }
+        else -> {
+          result.notImplemented();
+          return;
+        }
+      }
+    }catch (e : Exception){
+      println("OnMethodCall error "+ e);
     }
+    result.success(null);
```

Comment:
This has to be different on the basis of execution of different method. Like result.error, result.success(true/false). Take reference from hyper-sdk-flutter.

Reply:
actually `result.success(null);` is just for preventing flutter from throwing any error that `native channel didnt give any result back`

actually we can't get result back from native function when called from js interface provided by webview of flutter... its non blocking... so from upi intent i will call HyperWebviewBridge.postMessage(...) -> flutter will receive it and call native -> native again calls a flutter channel method (onActivityresult) and post the message back... now that flutter code will give the result back to upi intent ....